Framework/Kernel/src/DateAndTime.cpp

```diff
@@ -379,32 +371,10 @@ const DateAndTime &DateAndTime::defaultTime() {
 /** Sets the date and time using an ISO8601-formatted string
  *
  * @param str :: ISO8601 format string: "yyyy-mm-ddThh:mm:ss[Z+-]tz:tz"
- * @param displayLogs :: flag to indiciate if the logs should be displayed
  */
-void DateAndTime::setFromISO8601(const std::string &str, bool displayLogs) {
+void DateAndTime::setFromISO8601(const std::string &str) {
   // Make a copy
   std::string time = str;
```

Comment:
We could change the signature of `setFromISO8601` to `setFromISO8601(std::string time)` and remove the manual copy? That way we let the compiler do a move if possible: https://stackoverflow.com/questions/46028709/why-is-passing-by-value-if-a-copy-is-needed-recommended-in-c11-if-a-const-re

Reply:
`str` is used within the exception handling (which is why it is preserved)

Reply:
Oh yes, I missed that. In that case we should stick with the original `const &` version to avoid having 2 copies when an l-value input is used.

Either that or we remove the `str` from the exception and just throw the `std::invalid_argument` without including the string. The caller could then decide whether to include it in their error message if they want. It seems a shame that most of the time when the string is valid we are paying the price of a copy just for the error message.